compaction/compaction.cc

```diff
@@ -137,12 +137,18 @@ std::string_view to_string(compaction_type_options::scrub::quarantine_mode quara
 
 static api::timestamp_type get_max_purgeable_timestamp(const table_state& table_s, sstable_set::incremental_selector& selector,
         const std::unordered_set<shared_sstable>& compacting_set, const dht::decorated_key& dk, uint64_t& bloom_filter_checks,
-        const api::timestamp_type compacting_max_timestamp) {
+        const api::timestamp_type compacting_max_timestamp, const bool gc_check_only_compacting_sstables) {
     if (!table_s.tombstone_gc_enabled()) [[unlikely]] {
         return api::min_timestamp;
     }
 
     auto timestamp = api::max_timestamp;
+    if (gc_check_only_compacting_sstables) [[unlikely]] {
+        // If gc_check_only_compacting_sstables is enabled, do not
+        // check memtables and other sstables not being compacted.
+        return timestamp;
```

Comment:
Shouldn't we return `api::min_timestamp` in this case?

Reply:
cc @raphaelsc

Reply:
it's tricky, but it's actually max, since we want every timestamp to be eligible. minimum means nothing can be gced.